### PR TITLE
Split up rust-proto-new tests into pipeline grpc tests; proto_serde_tests; test_utils

### PR DESCRIPTION
**Pull request by wimax-grapl**
self explanatory. Nothing new, just reorganized.

### CODE REVIEW

File: src/rust/rust-proto-new/tests/pipeline_ingress_grpc_tests.rs [hunk 1]
```diff
@@ -0,0 +1,194 @@
+use futures::channel::oneshot::Sender;
```
Comment (wimax-grapl) [inline src/rust/rust-proto-new/tests/pipeline_ingress_grpc_tests.rs:1]: ^ the above doesn't *need* to be named 'integration_test.rs'

File: src/rust/rust-proto-new/tests/test_utils/strategies.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+use bytes::Bytes;
```
Comment (wimax-grapl) [inline src/rust/rust-proto-new/tests/test_utils/strategies.rs:1]: We can totally add more strategies here, I'm mostly doing this PR as a proof-of-concept / "hey here's how you'd add more stuff without siloing it to one file"

File: src/rust/rust-proto-new/tests/test_utils/strategies.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+use proptest::prelude::*;
+use uuid::Uuid;
+
```
Comment (jgrillo-grapl) [inline src/rust/rust-proto-new/tests/test_utils/strategies.rs:4]: Let's put the rest of the strategies in here too

File: src/rust/rust-proto-new/tests/proto_serde_tests.rs [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    }
+}
+
+//
+// ---------------- helpers ---------------------------------------------------
+//
+
+// helper function to define a simple encode-decode invariant
+// see: https://hypothesis.works/articles/encode-decode-invariant/
+fn check_encode_decode_invariant<T>(serializable: T)
```
Comment (jgrillo-grapl) [inline src/rust/rust-proto-new/tests/proto_serde_tests.rs:143]: This function should go in the test_utils

**Review by jgrillo-grapl — CHANGES_REQUESTED** [review]
A few more things and this will be ready to go

**Review by jgrillo-grapl — APPROVED** [review]
Didn't mean to make my previous comments blocking, this looks fine as is

File: src/rust/rust-proto-new/tests/proto_serde_tests.rs [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        },
+    },
+    SerDe,
+};
+use test_utils::{
+    serde::check_encode_decode_invariant,
+    strategies,
+};
+
+fn envelopes<T>(inner_strategy: impl Strategy<Value = T>) -> impl Strategy<Value = Envelope<T>>
```
Comment (jgrillo-grapl) [inline src/rust/rust-proto-new/tests/proto_serde_tests.rs:26]: these strategies should move too

File: src/rust/rust-proto-new/tests/proto_serde_tests.rs [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        created_time in any::<SystemTime>(),
+    ) -> PublishRawLogResponse {
+        PublishRawLogResponse {
+            created_time,
+        }
+    }
+}
+
+//
+// ---------------- helpers ---------------------------------------------------
```
Comment (jgrillo-grapl) [inline src/rust/rust-proto-new/tests/proto_serde_tests.rs:73]: this comment can go

**Review by jgrillo-grapl — APPROVED** [review]

**Pull request merged**